### PR TITLE
[Manual Backport 0.5-nexus] Add read/write bytes metrics (#803)

### PR DESCRIPTION
**Pull request by ykmr1224**
Manually backport https://github.com/opensearch-project/opensearch-spark/commit/2a647d43264cdcbb65f95b22c8373dee50497937 from https://github.com/opensearch-project/opensearch-spark/pull/803

Auto backport failed due to conflict in build.sbt

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -120,6 +120,7 @@ lazy val flintCore = (project in file("flint-core"))
       "software.amazon.awssdk" % "auth-crt" % "2.28.10",
       "com.fasterxml.jackson.core" % "jackson-core" % jacksonVersion,
       "com.fasterxml.jackson.core" % "jackson-databind" % jacksonVersion,
+      "org.projectlombok" % "lombok" % "1.18.30" % "provided",
```
Comment (noCharger) [inline build.sbt:123]: this is added in flintCommons, do we still need it in flintCore?

**Review by noCharger — APPROVED** [review]

File: build.sbt (hunk 1 above)
Comment (ykmr1224) [reply build.sbt:123]: I used lombok for this PR in flintCore.

Comment (noCharger) [reply build.sbt:123]: i mean flintCore depends on flintCommons, do we still need it?

Comment (seankao-az) [reply build.sbt:123]: if we want to remove it we can first remove in main. Let's keep this pr just as a simple backport

**Review by seankao-az — APPROVED** [review]

**Pull request merged**